src/plugins/gui/providers/moonpayProvider.ts

```diff
@@ -295,16 +321,30 @@ export const moonpayProvider: FiatProviderFactory = {
           if (exchangeAmount < minFiat)
             throw new FiatProviderError({ providerId, errorType: 'underLimit', errorAmount: minFiat, displayCurrencyCode: displayFiatCurrencyCode })
           // User typed a fiat amount. Need a crypto value
-          amountParam = `baseCurrencyAmount=${params.exchangeAmount}`
+          if (direction === 'buy') {
+            amountParam = `baseCurrencyAmount=${params.exchangeAmount}`
+          } else {
+            // Moonpay API doesn't let us specify a fiat amount for sell
+            throw new FiatProviderError({ providerId, errorType: 'paymentUnsupported' })
+          }
         } else {
           if (exchangeAmount > maxCrypto) throw new FiatProviderError({ providerId, errorType: 'overLimit', errorAmount: maxCrypto, displayCurrencyCode })
           if (exchangeAmount < minCrypto) throw new FiatProviderError({ providerId, errorType: 'underLimit', errorAmount: minCrypto, displayCurrencyCode })
-          amountParam = `quoteCurrencyAmount=${params.exchangeAmount}`
+          if (direction === 'buy') {
+            amountParam = `quoteCurrencyAmount=${params.exchangeAmount}`
+          } else {
+            amountParam = `baseCurrencyAmount=${params.exchangeAmount}`
+          }
         }
 
         const fiatCode = removeIsoPrefix(params.fiatCurrencyCode).toLowerCase()
-        const paymentMethod = useIAch ? 'ach_bank_transfer' : 'credit_debit_card'
-        const url = `https://api.moonpay.com/v3/currencies/${cryptoCurrencyObj.code}/buy_quote/?apiKey=${apiKey}&quoteCurrencyCode=${cryptoCurrencyObj.code}&baseCurrencyCode=${fiatCode}&paymentMethod=${paymentMethod}&areFeesIncluded=true&${amountParam}`
+        let url
+        if (direction === 'buy') {
+          url = `https://api.moonpay.com/v3/currencies/${cryptoCurrencyObj.code}/buy_quote/?apiKey=${apiKey}&quoteCurrencyCode=${cryptoCurrencyObj.code}&baseCurrencyCode=${fiatCode}&paymentMethod=${paymentMethod}&areFeesIncluded=true&${amountParam}`
+        } else {
+          url = `https://api.moonpay.com/v3/currencies/${cryptoCurrencyObj.code}/sell_quote/?apiKey=${apiKey}&quoteCurrencyCode=${fiatCode}&payoutMethod=${paymentMethod}&areFeesIncluded=true&${amountParam}`
+        }
```

Comment:
Similar suggestion:

```ts
const url =
  direction === 'buy'
    ? `https://api.moonpay.com/v3/currencies/${cryptoCurrencyObj.code}/buy_quote/?apiKey=${apiKey}&quoteCurrencyCode=${cryptoCurrencyObj.code}&baseCurrencyCode=${fiatCode}&paymentMethod=${paymentMethod}&areFeesIncluded=true&${amountParam}`
    : `https://api.moonpay.com/v3/currencies/${cryptoCurrencyObj.code}/sell_quote/?apiKey=${apiKey}&quoteCurrencyCode=${fiatCode}&payoutMethod=${paymentMethod}&areFeesIncluded=true&${amountParam}`;
```

This isn't too terse for a ternary.

Reply:
Hmm. I think it's a toss up on this one. When it's a cointoss, I lean towards code where I can set break points